pkg/controllers/workgenerator/controller.go

```diff
@@ -180,24 +183,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 			Message:            "All of the works are synchronized to the latest",
 		})
 		if workUpdated {
-			// revert the applied condition and failedPlacement if we made any changes to the work
-			resourceBinding.Status.FailedPlacements = nil
+			// revert the applied condition if we made any changes to the work
```

Comment:
Why only revert the applied? Where are applied and available set in this case? It seems they are both not there.

Reply:
we set the applied condition as false below. In that case, available condition nil means unknown.

In the CRP status, if the previous condition is not true, we will reset the remaining and leave them as "unknown".

Reply:
why set it to false then?  it seems that leaving it not set has the same effect?